### ISSUE TITLE
accessing package specific functions in other layers

### ISSUE THREAD

**Issue by Luis-Henriquez-Perez** [ekaschalk/dotspacemacs#5]
I've been analyzing and emulating your layer setup for spacemacs configuration.

I've noticed that you added macros you defined in the macros layer's ` funcs.el` file.
What if I wanted to install a package with macros instead?
Two of the macros you use -- the macro for an anonymous function `xi` and `evil-define-keys`-- have already been written. 
`https://github.com/noctuid/general.el`
`https://github.com/troyp/fn.el`
If instead of writing those macros you added these packages to the macros layer, how would you make the macros in these packages available for other layers?

Other layers can't see the methods of installed packages in the `macros` layer, they can only see the functions in its `funcs.el`

Right now I the only way I can think of doing that is copying the functions from the packages and pasting them in to `funcs.el` in the macros layer. Do you know of a better way?

**Comment by ekaschalk** [comment]
Setting up the package in `packages.el` and requiring the layer in the others didn't work? I install `dash-functional` in macros/packages.el and use its api throughout the rest of my layers.

**Comment by Luis-Henriquez-Perez** [comment]
No. I keep getting an error that the package function is void.

To be honest for me even using the defining functions in the `macros` layer's `funcs.el` and requiring it in other layers didn't work for me. 
I keep getting the error 
`An error occurred while post-configuring evil in layer config (error: (void-function --each))`.
So weirdly it did know the macros layer defined a macro called `evil-global-set-keys`  but it did not know about `dash-functional`.

I'll put my config on github in case you want to take a look. `https://github.com/timidpiglet/.spacemacs.d`

Also I've noticed that generally on the first time installing dash-functional, it works. But every time after that it does not.

**Comment by timidpiglet** [comment]
I'll put my config on github in case you want to take a look.

**Comment by Luis-Henriquez-Perez** [comment]
I got `global-evil-set-keys` to work by wrapping any keys I cast with `with-eval-after-load 'dash`. 
I got the `general` package to work by wrapping keys I cast with `with-eval-after-load 'general`.
It's slightly more inconvenient but it's a start.

I'm still uncertain as to why my loading is working differently than yours. And why it was the `dash` package as opposed to `dash-functional` that was not loading in time for `config/post-init-evil`.

I'm not sure what I did or could have done in my code for this to have come about. I will push my modifications on my `.spacemacs.d`.
